src/template-tags/tickets.php

```diff
@@ -403,6 +396,48 @@ function tribe_events_ticket_is_on_sale( Tribe__Tickets__Ticket_Object $ticket )
 	}
 }//end if
 
+if ( ! function_exists( 'tribe_tickets_is_current_time_in_date_window' ) ) {
+
+	/**
+	 * Checks if the post has tickets that are available in the current date range set on the ticket.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $post_id Post (or event) to check for ticket availability.
+	 *
+	 * @return bool
+	 */
+	function tribe_tickets_is_current_time_in_date_window( $post_id ) {
+		static $ticket_availability = [];
+
+		if ( isset( $ticket_availability[ $post_id ] ) ) {
+			return $ticket_availability[ $post_id ];
+		}
+
+		$has_tickets_available = false;
+		$tickets               = Tribe__Tickets__Tickets::get_all_event_tickets( $post_id );
+		$default_provider      = Tribe__Tickets__Tickets::get_event_ticket_provider( $post_id );
+
+		foreach ( $tickets as $ticket ) {
+			$ticket_provider = $ticket->get_provider();
+
+			// Skip tickets that are for a different provider than the event provider.
+			if (
+				$default_provider !== $ticket_provider->class_name
+				&& Tribe__Tickets__RSVP::class !== $ticket_provider->class_name
+			) {
+				continue;
+			}
+
+			$has_tickets_available = ( $has_tickets_available || tribe_events_ticket_is_on_sale( $ticket ) );
```

Comment:
This is going to run through all the tickets no matter what, but it looks like you can break out once this is set to true?

Reply:
```suggestion
			if ( tribe_events_ticket_is_on_sale( $ticket ) ) {
			    $has_tickets_available = true;
			    break;
			}
```

You could even shorten it a bit more by leaving out `$has_tickets_available` and just setting `$ticket_availability[ $post_id ]` ?

Reply:
`tribe_tickets_is_current_time_in_date_window()` came from @borkweb in https://github.com/moderntribe/event-tickets/pull/1533/commits/e21e6c2fd8769983ddb72959b1030784c469003a so I didn't want to affect it for eventual merging with B20.01/G20.01

But good suggestions so let's do it anyway :)

Reply:
I added a note for myself to come back to this function in my optimization tasks. We can keep as is for now.